Fails to build on SL6.9 because of old unity version C99 incompatibility

*Please use this template for reporting suspected bugs or requests for help.*

# Issue description

Build fails with error message because of C99 style variable initialisation in bundled unity framework.

# Environment

* libzmq version (commit hash if unreleased):
commit a8d19057b109ce637aa19cda1a3c6d3402143a3a
* OS: 
Scientific Linux 6.9
* Compiler:
gcc 4.4.7

# Minimal test code / Steps to reproduce the issue

1. Compilation fails 


# What's the actual result? (include assertion message & call stack if applicable)

[ 49%] Building C object tests/CMakeFiles/unity.dir/__/external/unity/unity.c.o
/home/AMS/drasiams/libzmq/external/unity/unity.c: In function ‘UnityParseOptions’:
/home/AMS/drasiams/libzmq/external/unity/unity.c:1408: error: ‘for’ loop initial declarations are only allowed in C99 mode
/home/AMS/drasiams/libzmq/external/unity/unity.c:1408: note: use option -std=c99 or -std=gnu99 to compile your code



# What's the expected result?

Compilation should succeed.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,6 +67,8 @@ endif()
 check_c_compiler_flag("-std=gnu11" COMPILER_SUPPORTS_C11)
 if(COMPILER_SUPPORTS_C11)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu11")
+else()
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
 endif()
 
 if(NOT MSVC)

--- a/RELICENSE/bl0x.md
+++ b/RELICENSE/bl0x.md
@@ -1,0 +1,16 @@
+# Permission to Relicense under MPLv2 or any other OSI approved license chosen by the current ZeroMQ BDFL
+
+This is a statement by Bastian Löher that grants permission to
+relicense its copyrights in the libzmq C++ library (ZeroMQ) under the
+Mozilla Public License v2 (MPLv2) or any other Open Source Initiative
+approved license chosen by the current ZeroMQ BDFL (Benevolent
+Dictator for Life).
+
+A portion of the commits made by the Github handle "bl0x", with
+commit author "Bastian Löher <me@l-dot.de>", are
+copyright of Bastian Löher.  This document hereby grants the libzmq
+project team to relicense libzmq, including all past, present and
+future contributions of the author listed above.
+
+Bastian Löher
+2019/08/01

